code/modules/telesci/dses.dm

```diff
@@ -0,0 +1,291 @@
+/*Deep Space Exploration System
+	used like a sonar system, you ping it, it returns if it finds something with a ping, else a buzz
+
+	POTENTIAL MODULES
+
+ - Ping resonation locator
+	-Tells the user the direction of the caught turf
+ - Ping resonation logging
+	-Logs the co-ordinates of the caught turf
+ - ping resource resequencing
+	-Pings now cost half
+ - ping long-range listener
+	-Ping range doubled
+ - Auto ping system
+	-DSES ping is now togglable, if on it pings every 1 SECOND
+
+*/
+
+/obj/item/device/dses
+	name = "deep space exploration system"
+	desc = "A GPS with a high-gain radio antenna and broadcaster for locating proximity objects in space, the explorers friend."
+	icon_state = "dses"
+	icon = 'icons/obj/telescience.dmi'
+	var/obj/item/weapon/cell/C = null
+	var/module_limit = 1
+	var/list/module_list = list()
+	var/list/locations = list()
+	var/pulse_range = 25
+	var/pulse_cost = 250
+	var/pulse_between = 3 SECONDS
+	var/last_pulse
+	var/last_direction
+	var/last_distance
+	var/list/positive_locations = list()
+	var/auto_pulse = 0
+
+/obj/item/device/dses/attack_self(var/mob/user as mob)
+	if(C)
+		return menu_open(user)
+	else
+		to_chat(user, "<span class = 'notice'>The screen remains dark.</span>")
```

Comment:
Working on the premise that the battery would conserve at least some internal power to support critical systems, such as the screen.

No battery, no screen.

No charge, no pulsing.